java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -771,41 +806,34 @@ private void buildTryStatement(TryStatementTree tryStatementTree) {
     }
     Block finallyOrEndBlock = currentBlock;
     Block beforeFinally = createBlock(currentBlock);
-    List<Block> catches = new ArrayList<>();
+    Map<Type, Block> catches = new HashMap<>();
+    TryStatement tryStatement = new TryStatement();
+    tryStatement.successorBlock = finallyOrEndBlock;
+    enclosingTry.push(tryStatement);
+    enclosedByCatch.push(false);
```

Comment:
If the `TryStatementTree` is within a catch block, then it will be wrong. We should peek the top of the stack.